### PR TITLE
Update why ccf + Implement application claim for transfer

### PR DESCRIPTION
**Pull request by takuro-sato**

### CODE REVIEW

File: banking-app/README.md [hunk 1]
```diff
@@ -42,7 +42,10 @@ If you want implement an ATM, the deposit procedure would be:
 
 There are multiple reasons that you want to use CCF for banking consortium.
 
-TODO
+- The ledger generated by CCF is tamper proof and banks can prove that there is no tamper in the ledger.
```
Comment (achamayou) [inline banking-app/README.md:45]:
```suggestion
- The ledger generated by CCF is tamper-proof, it cannot be modified after having been produced. Even users without a full copy of the ledger, holding their transaction receipts, are protected.
```

File: banking-app/README.md [hunk 2]
```diff
@@ -42,7 +42,10 @@ If you want implement an ATM, the deposit procedure would be:
...
+- Users can use [transaction receipts](https://microsoft.github.io/CCF/main/audit/receipts.html#receipts) with [application claim](https://microsoft.github.io/CCF/main/use_apps/verify_tx.html#application-claims) as proof of transferring money even without accessing to the system (TODO: implement application-claims in the app).
```
Comment (achamayou) [inline banking-app/README.md:46]:
```suggestion
- Users can use [transaction receipts](https://microsoft.github.io/CCF/main/audit/receipts.html#receipts) with [application claim](https://microsoft.github.io/CCF/main/use_apps/verify_tx.html#application-claims) as proof of transferring money even without online access to the system (TODO: implement application-claims in the app).
```

File: banking-app/README.md [hunk 3]
```diff
@@ -42,7 +42,10 @@ If you want implement an ATM, the deposit procedure would be:
...
+- Banks don't have to trust each other. Majority of the banks need to agree to make changes on the application with the default [constitution](https://microsoft.github.io/CCF/main/governance/constitution.html#constitution).
```
Comment (achamayou) [inline banking-app/README.md:47]:
```suggestion
- Banks don't have to trust each other. A majority of the banks need to agree to make changes on the application with the default [constitution](https://microsoft.github.io/CCF/main/governance/constitution.html#constitution).
```

File: banking-app/test.sh [hunk 4]
```diff
@@ -57,6 +57,19 @@ check_eq "Transfer: 40 from user0 to user1" "204" "$(curl $server/app/transfer/$
 check_eq "Balance: user0, account_type0" "{\"balance\":60}" "$(curl $server/app/balance/$account_type0 -X GET $(cert_arg "user0") -s)"
 check_eq "Balance: user1, account_type1" "{\"balance\":40}" "$(curl $server/app/balance/$account_type1 -X GET $(cert_arg "user1") -s)"
 
+# Test reciept
```
Comment (achamayou) [inline banking-app/test.sh:60]:
```suggestion
# Test receipt
```

File: banking-app/test.sh [hunk 5]
```diff
@@ -57,6 +57,19 @@ check_eq "Transfer: 40 from user0 to user1" "204" "$(curl $server/app/transfer/$
...
+transfer_transaction_id=$(curl $server/app/transfer/$account_type0 -X POST $(cert_arg "user0") -H "Content-Type: application/json" --data-binary "{ \"value\": 5, \"user_id_to\": \"$user1_id\", \"account_name_to\": \"$account_type1\" }" -i -s | grep -i x-ms-ccf-transaction-id | awk '{print $2}' | sed -e 's/\r//g')
+# Wait for reciept to be ready
```
Comment (achamayou) [inline banking-app/test.sh:62]:
```suggestion
# Wait for receipt to be ready
```

File: banking-app/app.json [hunk 6]
```diff
@@ -107,6 +107,30 @@
           }
         }
       }
+    },
+    "/claim": {
```
Comment (achamayou) [inline banking-app/app.json:111]: /receipt

File: banking-app/app.json [hunk 7]
```diff
@@ -107,6 +107,54 @@
...
+    },
+    "/receipt": {
+      "get": {
+        "js_module": "endpoints/banking.js",
+        "js_function": "getTransactionReceipt",
+        "forwarding_required": "always",
+        "authn_policies": ["user_cert"],
+        "mode": "readonly",
+        "openapi": {
+          "responses": {
```
Comment (takuro-sato) [inline banking-app/app.json:143]: example:
```json
{
  "cert": "-----BEGIN CERTIFICATE-----\nMIIBwjCCAUigAwIBAgIQFzAiH4NkcLfv2PCzZBD79DAKBggqhkjOPQQDAzAWMRQw\nEgYDVQQDDAtDQ0YgTmV0d29yazAeFw0yMjEwMjQxMDUzMDhaFw0yMzAxMjIxMDUz\nMDdaMBMxETAPBgNVBAMMCENDRiBOb2RlMHYwEAYHKoZIzj0CAQYFK4EEACIDYgAE\nERZww/iGKK8LtgQpvaa7jKYItenps7FXPk8stGcl6QmRgwQKEBp+IgV7/dfOOA8c\nZR/24miP1nCk6cK22CI9DBtUl4l8P8h4tot/tyv7e4P3Qr7sCi3L+MZBoF0tb4ys\no14wXDAJBgNVHRMEAjAAMB0GA1UdDgQWBBStTaoXnkvNvZyAEGwziWMRnFJrYjAf\nBgNVHSMEGDAWgBRFTSi8o04oFUdStEf568jR6SZrHDAPBgNVHREECDAGhwR/AAAB\nMAoGCCqGSM49BAMDA2gAMGUCMG38Z5P9/oYJBum/HShEDTly6VyrcNG4AkN2lCE2\nK59Ma9rsku9Tms5pfsoL90OUTAIxAJVRa57GAD5fdlaZJsUomeI0s5Zh7+AJVr//\n6yGmlnjJlvbJfiV58FW+tv9LNZY2pQ==\n-----END CERTIFICATE-----\n",
  "leaf_components": {
    "claim": "2a278a3b8382b86fecf5c1af4f04d40dafa9293fd5a5b8837297e9ae923b3bd4 sent 40 to 348edea0724545b78ed642e2e9c4393c871f30cc54b1e35f5b8a28b4a9abf745",
    "commit_evidence": "ce:2.33:c597d6f3066efd7852576a2e5f32ab1a6685eec1058f5764517c8acbdbdda923",
    "write_set_digest": "47483850c9d3c7a57d9aae1d7006d71936359c4ef54e5a20ce847de3a707fa09"
  },
  "node_id": "7dbecf20df9c14887471c95bf42765106eee68890959acc50f2f1a48aa5b3e4c",
  "proof": [
    {
      "left": "16386d598426db141f61b81e5e428dba4b2ddb6989b5cc2c98eb59533903f969"
    },
    {
      "left": "48caa55b5d0c4b6b442f9d21a28a34df8d84aaf604db45e16ad9b088809f74c3"
    }
  ],
  "signature": "MGUCMDai3nW1ks1PL1lP55jTyr07UzzrOzGfJG69pgprYJkehe/WpyThsfq6Ef9pwS94RwIxALBkung8R0inFX4My/9iQqvaaZR0uOheZW9LnooiXVs+lS5a+Pf9Qk/FYd4STZizMA=="
}
```

File: banking-app/src/endpoints/banking.ts [hunk 8]
```diff
@@ -175,27 +187,221 @@ export function transfer(
...
     statusCode: 204
   };
 }
+
+interface GetClaimResponse {
+  claim: string
+}
+
+function validateTransactionId(transactionId: any): boolean {
+  // Transaction ID is composed of View ID and Sequesnce Number
```
Comment (achamayou) [inline banking-app/src/endpoints/banking.ts:224]:
```suggestion
  // Transaction ID is composed of View ID and Sequence Number
```

File: banking-app/src/endpoints/banking.ts [hunk 9]
```diff
@@ -175,27 +187,221 @@ export function transfer(
...
+  }
+
+  const userId = getCallerId(request);
+  const txNums = transactionId.split('.');
+  const seqno = parseInt(txNums[1]);
+
+  const rangeBegin = seqno;
+  const rangeEnd = seqno;
+
+  // Make hundle based on https://github.com/microsoft/CCF/blob/main/samples/apps/logging/js/src/logging.js
```
Comment (achamayou) [inline banking-app/src/endpoints/banking.ts:257]:
```suggestion
  // Make handle based on https://github.com/microsoft/CCF/blob/main/samples/apps/logging/js/src/logging.js
```

**Review by achamayou — APPROVED** [review]
1. remove /claim
2. update test to use /receipt

**Pull request merged**